### PR TITLE
[java] Alert Listener implementation

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
- [x] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

Alert Listener Implementation
```
beforeAlertAccept
afterAlertAccept
afterAlertDismiss
beforeAlertDismiss
```

@lukeis please review

### CODE REVIEW

File: java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java [hunk 1]
```diff
@@ -63,7 +64,7 @@
  * {@link WebDriverEventListener}, e&#46;g&#46; for logging purposes.
  */
 public class EventFiringWebDriver implements WebDriver, JavascriptExecutor, TakesScreenshot,
-    WrapsDriver, HasInputDevices, HasTouchScreen {
+    WrapsDriver, HasInputDevices, HasTouchScreen, Alert {
```
Comment (SrinivasanTarget) [inline java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java]: @lukeis I see tests failed here: https://travis-ci.org/SeleniumHQ/selenium/jobs/173262596

Can you help me in resolving the same?

Comment (lukeis) [inline java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java]: your Mock of WebDriver needs to respond to driver.switchTo() and return a mock of a TargetLocator and then the targetLocator.alert() needs to return your mocked Alert.

Comment (SrinivasanTarget) [inline java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java]: @lukeis Can you please check?

Comment (SrinivasanTarget) [inline java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java]: @lukeis Sorry to bother you. I'm not getting the failures actually.

**Comment by SrinivasanTarget** [comment]
@lukeis Have fixed it. Travis is green here: https://travis-ci.org/SeleniumHQ/selenium/jobs/173961677

**Comment by SrinivasanTarget** [comment]
@lukeis Failures were not in java code. Travis is Green.

**Comment by SrinivasanTarget** [comment]
@lukeis Can you please review and merge this?

**Comment by SrinivasanTarget** [comment]
@lukeis Thanks :)

**Pull request merged**